worker-message-prioritization-rerouting/src/main/java/com/github/workerframework/workermessageprioritization/rerouting/StagingQueueCreator.java

```diff
@@ -32,17 +37,28 @@ public StagingQueueCreator(final Channel channel) {
     }
 
     void createStagingQueue(final Queue targetQueue, final String stagingQueueName) throws IOException {
-        
+
         if(declaredQueues.contains(stagingQueueName)) {
             return;
         }
 
-        channel.queueDeclare(
-            stagingQueueName,
-            targetQueue.isDurable(),
-            targetQueue.isExclusive(),
-            targetQueue.isAuto_delete(),
-            targetQueue.getArguments());
+        final boolean durable = targetQueue.isDurable();
+        final boolean exclusive = targetQueue.isExclusive();
+        final boolean autoDelete = targetQueue.isAuto_delete();
+        final Map<String, Object> arguments = targetQueue.getArguments();
+
+        LOGGER.info("Creating or checking staging queue by calling channel.queueDeclare({}, {}, {}, {}, {})",
```

Comment:
This INFO log is to try and help diagnose https://internal.almoctane.com/ui/entity-navigation?p=131002/6001&entityType=work_item&id=625207. What happens is:

1) A staging queue gets created (`channel.queueDeclare` ) **without** an x-max-priority arg

2) Sometime later, maybe after the worker has been restarted and the in-memory `declaredQueues` set in this class is empty, `channel.queueDeclare` is called again for the same staging queue, but the second time it **does** have an x-max-priority. Rabbit does not allow this, and an exception will be thrown from `channel.queueDeclare`

This INFO log will output the queue arguments for the target queue that were returned by the RabbitMQ API. It will tell us if the API is returning the queue without an x-max-priority arg, in which case we can assume that we cannot trust the response the from API, and will need to create the queues another way (static config files or something).

Reply:
> This INFO log is to try and help diagnose https://internal.almoctane.com/ui/entity-navigation?p=131002/6001&entityType=work_item&id=625207. What happens is:
> 
> 1. A staging queue (`dataprocessing-family-hashing-in»/rory3/ingestion-workflow`) gets created (`channel.queueDeclare` ) **without** an x-max-priority arg
> 2. Sometime later, maybe after the worker has been restarted and the in-memory `declaredQueues` set in this class is empty, `channel.queueDeclare` is called again for the same staging queue, but the second time it **does** have an x-max-priority. Rabbit does not allow this, and an exception will be thrown from `channel.queueDeclare`
> 
> This INFO log will output the queue arguments for the target queue (`dataprocessing-family-hashing-in`) that were returned by the RabbitMQ API. It will tell us if the API is returning the target queue without an x-max-priority arg, in which case we can assume that we cannot trust the response from the RabbitMQ API, and will need to create the staging queues another way (static config files or something).
> 
> The problem is, we only find out there is a problem when (2) happens (i.e the worker is restarted and tries to declare the same staging queue but with different args).
> 
> My thinking is that if (2) happens, we will be able to view the older logs to see this new INFO log, to see if the queue was initially created without an x-max-priority field. If so, we can assume we cannot trust the RabbitMQ API response.
> 
> I am hoping the INFO log will still be there even if the worker is restarted. I think it should be?
> 
> I made this an INFO log as it will only be logged the first time a worker creates a staging queue, or the first time a worker has seen a staging queue (e.g. after a worker restart), so it will not spam the logs, i.e. it should be infrequent.
> 
> It could be a DEBUG log, but then we'd have to run all the workers at DEBUG level during testing, which will result in huge logs.
> 
> @michael-bryson Does this approach sound ok with you? It means we would just have to run the workers involved in rerouting at INFO level when we're testing this.

@rorytorneymf Yes this approach seems fine to me, this will only be used during testing purposes and if we find the cause of the issue and handle it then we can change the level to debug for production.